src/api/posts.js

```diff
@@ -0,0 +1,64 @@
+const BASE_URL = 'https://mate-api.herokuapp.com';
+
+export async function getUserPosts(userId) {
+  const posts = await fetch(`${BASE_URL}/posts`)
+    .then(response => response.json())
+    .then(result => result.data);
+
+  if (userId === 0) {
+    return posts;
+  }
+
+  return posts.filter(post => post.userId === userId);
+}
+
+export async function getPostDetails(postId) {
+  const data = await fetch(`${BASE_URL}/posts/${postId}`)
+    .then(response => response.json())
+    .then(result => result.data);
+
+  return data;
+}
+
+export async function getPostComments(postId) {
+  const comments = await fetch(`${BASE_URL}/comments`)
+    .then(response => response.json())
+    .then(result => result.data);
+
+  return comments.filter(comment => comment.postId === postId);
+}
+
```

Comment:
according to the task you should move all responses connected with comments to separate file comments.js)